magellan-sample-migration/src/test/java/com/wealthfront/magellan/sample/migration/tide/DogListStepTest.kt

```diff
@@ -17,22 +16,20 @@ import org.junit.runner.RunWith
 import org.robolectric.Robolectric
 import org.robolectric.RobolectricTestRunner
 import org.robolectric.Shadows.shadowOf
-import javax.inject.Inject
 
 @RunWith(RobolectricTestRunner::class)
 class DogListStepTest {
-  @Inject lateinit var api: DogApi
 
   private var chosenBreed: String? = null
-  private val dogListStep = DogListStep { chosenBreed = it }
+  private lateinit var dogListStep: DogListStep
   private val activityController = Robolectric.buildActivity(ComponentActivity::class.java)
 
   @Before
   fun setUp() {
     val context = ApplicationProvider.getApplicationContext<Application>()
-    ((context as AppComponentContainer).injector() as TestAppComponent).inject(this)
-
-    coWhen { api.getAllBreeds() }
+    val component = ((context as AppComponentContainer).injector() as TestAppComponent)
+    dogListStep = DogListStep(component.toolbarHelper, component.api) { chosenBreed = it }
```

Comment:
Instead of exposing individual services on TestAppComponent, what do you think about having Dagger inject DogListStepTest with a DogListStepFactory?